packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-case-custom-pwd-recovery-mfa/main/java/siw.md

```diff
@@ -0,0 +1,122 @@
+### 1: Start password recovery
+
+The user starts the password recovery flow with the following steps:
+
+1. Clicks the **Forgot Password?** link on the sign-in page.
+2. Enters their **Email or Username** in the dialog and then clicks **Next**.
+3. Chooses **Email** as the authenticator they want to use for password recovery and clicks **Submit**.
+
+Okta then tells the user to either click the link in the email or enter the code to continue and sends an email to their email address matching the Forgot Password template that was altered earlier.
+
+<div class="common-image-format">
+
+![Example of email sent to user](/img/advanced-use-cases/custom-pwd-recovery-custom-email.png "Password recovery email")
+
+</div>
+
+The email's **Reset Password** link includes the `otp` and `request.relayState` variables sent back as query parameters to the application. For instance, the URL in the email template,  `http://localhost:8080/magic-link/callback?otp=${oneTimePassword}&state=${request.relayState}`, might be rendered as `http://localhost:8080/magic-link/callback?otp=726009&state=1b34371af02dd31d2bc4c48a3607cd32` in the email sent to the user.
+
+### 2: Handle the OTP and state parameters
+
+Create a callback handler method that takes the `otp` and `state` parameters in the query string and passes them as session parameters to a page that contains the Sign-In Widget. Before calling the callback handler method, retrieve the current `IDXClientContext` object from the browser session and check that `state` has a value.
+
+If `state` is `null` or the browser context object is `null` (because the user clicked the magic link from a different browser), the widget instructs advise the user to return to the original tab in the browser where they requested a password reset and enter the OTP to proceed.
```

Comment:
See my comment above for the same use of these parameters. The current text is a bit confusing and isn't in the appropriate sequence.

Reply:
Changed to

> Create a callback handler that takes the `otp` and `state` parameters from the query string, attempts to retrieve the current `IDXClientContext` object from the browser session, and then passes them all as session parameters to a page that contains the Sign-In Widget for processing.
> 
> If either the browser context object is `null` (because the user clicked the magic link from a different browser) or `state` is `null`, redirect to an error page instead.

Reply:
This has the same issue as the previous one. Who calls the handler (and possible when)? What are the arguments for the handler (the current wording is unclear). The structure is also a bit misleading as it reads like: "createa a callback handler with no parameters that's called magically and extracts the otp and state parameters from the query string, then retrievs the ... etc."

BTY, "attempts to retrieve" is another "could" type expression. Either `IDXClientContext` is retrieved or it's not. If failure to retrieve it is an important error, then call it out.

Reply:
Changed to

> Create a mapping that maps a request for the magic link URL to a method that takes the `otp` and `state` parameters from the query string, and makes the following checks:
> 
> 1. That `otp` and `state` are not `null`.
> 2. That `state` matches the state stored in the current `ProceedContext` session variable.
> 
> If either check returns false, throw an error.

Idly, the concept of "attempting to do a thing" is baked into code isn't it? That's why they are called try..catch, and if...then clauses. Similarly here we have attempts to retrieve a value from several different locations and if the value isn't anywhere something else happens instead. I've changed it now in the text, but....